### PR TITLE
Accept handlers return closes #1704

### LINKED ISSUE #1704: improve multi handles on one message.
Currently nothing do on handler functions return.
but we can use to improve handle matching on function returns:
see this example:

```python3

from telebot.async_telebot import AsyncTeleBot
from telebot.return_types import ContinueHandling # new return_types

bot = AsyncTeleBot('TOKEN')
database = ...


@bot.message_handler(commands=['help'])
async def cmdlist(message):
    if message.chat.id not in database:
        await bot.reply_to(message, "Hello, I see you first time.")
    return ContinueHandling


@bot.message_handler(commands=['help'])
async def cmdlist(message):
    await bot.reply_to(message, "helpmsg")
```

### PR DESCRIPTION
**Pull request by byehack**
basic implement for https://github.com/eternnoir/pyTelegramBotAPI/pull/1704

- [ ] Test
- [ ] Create example
- [ ] Document
- [ ] Bump new version

### CODE REVIEW

**Comment by coder2020official** [comment]
Hi!
I'll investigate it and review ASAP. Thank you for pull request.

File: telebot/__init__.py [hunk 1]
```diff
@@ -5924,13 +5927,14 @@ def _run_middlewares_and_handler(self, message, handlers, middlewares, update_ty
                     if not process_handler: continue
                     for i in inspect.signature(handler['function']).parameters:
                         params.append(i)
+                    ret = None
```
Comment (coder2020official) [inline telebot/__init__.py:5930]: Renaming this variable to result would be more appropriate
```suggestion
                    result = None
```
And other variables should be renamed as well.

File: telebot/__init__.py [hunk 2]
```diff
@@ -5924,13 +5927,14 @@ def _run_middlewares_and_handler(self, message, handlers, middlewares, update_ty
...
                     if len(params) == 1:
-                        handler['function'](message)
+                        ret = handler['function'](message)
```
Comment (coder2020official) [inline telebot/__init__.py:5932]: Don't forget applying result instead of ret to async as well

File: telebot/handler_backends.py [hunk 3]
```diff
@@ -266,5 +261,10 @@ class CancelUpdate:
     of post_process in middlewares.
     """
 
-    def __init__(self) -> None:
-        pass
\ No newline at end of file
+
+class Continue:
```
Comment (coder2020official) [inline telebot/handler_backends.py:265]: Renaming this to ContinueHandling would be more appropriate.. What do you think?

**Review by coder2020official — CHANGES_REQUESTED** [review]

**Comment by Badiboy** [comment]
Also please update 
ret = 
to 
result = 
It's the general practice in this code.

**Comment by coder2020official** [comment]
Thank you for fixing the requested changes. We are now discussing about it with badiboy. We will let you know about the further changes.

**Comment by coder2020official** [comment]
After resolving #1722 , you will have to make little changes(we will notify you)

**Comment by coder2020official** [comment]
Btw, thank you for contributing and your patience.

**Comment by coder2020official** [comment]
#1722 was merged, can you take a look and adapt your code?

**Comment by coder2020official** [comment]
You deleted branch.. can you recreate it?

**Comment by byehack** [comment]
@coder2020official opened https://github.com/eternnoir/pyTelegramBotAPI/pull/1742

**Pull request closed without merging**